cli/klaus/src/main.rs

```diff
@@ -94,31 +126,61 @@ async fn main() -> Result<(), Error> {
     }
     register_activation(&args.registry_path, &args.dot_flox_hash, activation)?;
 
-    // Start the listeners
-    let shutdown_flag = Arc::new(AtomicBool::new(false));
-    let signal_listener = signal_listener()?;
-    let signal_task = spawn_signal_listener(signal_listener, shutdown_flag.clone())?;
-    let termination_task = spawn_termination_listener(activation.into(), shutdown_flag.clone());
-
     info!(
         this_pid = nix::unistd::getpid().as_raw(),
         target_pid = args.pid,
         "watchdog is on duty"
     );
 
     // Listen for a notification
-    let action = listen(signal_task, termination_task, shutdown_flag).await;
-
-    match action {
-        Action::Cleanup => {
-            info!(pid = &args.pid, "deregistering activation");
-            deregister_activation(&args.registry_path, &args.dot_flox_hash, activation)
-                .context("failed to deregister activation")?;
-        },
-        Action::Terminate => {
-            // TODO: deregister if !is_running?
```

Comment:
(non-blocking because this PR doesn't change the existing behaviour but I think we should followup with a separate PR or ticket)

I still feel cautious about _not_ deregistering on what is now `should_stop` because we may leave an environment in a position where services are never stopped automatically because a stray activation PID is left in the registry.

In hindsight, after writing the comment, checking whether the activation is still running would likely hit a race condition where it is always true if both processes received signals at the same time. However I do think it would be preferable to deregister but not stop `process-compose` and allow it to be cleaned up in future.

Reply:
I think the choice is between pulling the rug out from under the activation or potentially leaking services. When put that way, I think pulling the rug is preferable. Deregistering and not stopping `process-compose` seems like a problem because `process-compose` gets re-parented to PID 1, which will only clean it up on shutdown.

Reply:
https://github.com/flox/flox/issues/1904